### PR TITLE
LPS-189853 Refactor view_item_selector_view_descriptor after revert

### PR DESCRIPTION
**Pull request by BarbaraCabrera**
[Link to ticket](https://liferay.atlassian.net/browse/LPS-189853)

Hi again, @liferay-lima 👋🏼 

As @ambrinchaudhary and I discussed in the previous [pr thread](https://github.com/liferay-lima/liferay-portal/pull/4624), I leave the changes here.

Now this also fix the behaviour share by @Jeremy-Ch :

**Case1:** Cannot change the display style in item selector
1. Go to People> Memberships
2. Click “+“
3. Change the display style

**Case2:** Cannot change the site scope in the item selector
1. Go to add a blog entry
2. Select a related assets
3. Click the “Sites and Libraries“ breadcrumb in the item selector

The difference is that instead of making a `searchContainerOnHandler.dispose();` we need to make a `searchContainerOnHandler.detach();` in the ViewItemSelectorViewDescriptorMultiple.ts

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Review by ambrinchaudhary — APPROVED** [review]

**Comment by Tim-Cao** [comment]
ci:stop

**Comment by Tim-Cao** [comment]
ci:test:echo

**Comment by Jeremy-Ch** [comment]
From the Lima side, the failures are unrelated. It's good to forward.

**Comment by Tim-Cao** [comment]
Hi @BarbaraCabrera, the Showing 1 - 3 of 5 results. missing on Asset Publisher
Would you mind having a look at?
Steps to reproduce

1. Add 5 web contents
2. Add an Asset Publisher to a widget page
3. Open the configuraiton modal of Asset Publisher
4. Change asset selection as dynamic
5. Change the pagination type to regular
6. Change the Number of Items to Display to 3 then Save
7. Close the modal

Expected
image

Actual
image

**Comment by Tim-Cao** [comment]
Hi @BarbaraCabrera, no failures are related to your changes from ci:test:echo run

**Comment by BarbaraCabrera** [comment]
ci:forward
